### PR TITLE
PRIME-1974 Health Authority Admin Overview

### PR DESCRIPTION
**Pull request by hogmann**
Health authority admin updates.

**New Pages:**
- Site Overview Page
- Admin Notes
- Admin Events
- Admin Documents

**New Functionality:**
- HA Site Actions
- HA Site table rows have new data 
- Assign/Reassign
- Flag

**Behind the scenes**
- Original Site Overview page uses container component not extends
- Abstract Admin Site Class to make a single point for functionality
- Made generic site API for actions and removed the return of the whole site on a lot of the calls

**Testing**
- Health Authority Site Overview
- Health Authority Site Events
- Health Authority Site Documents
- Health Authority Site Notes
- Health Authority Site Actions
- Site Overview
- Site Actions

### CODE REVIEW

File: prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.html
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.html:7]: Separate ticket in next sprint and does not need to be evaluated

File: prime-angular-frontend/src/app/modules/adjudication/pages/site-overview/site-overview.component.ts
Comment (AMiKaH) [inline prime-angular-frontend/src/app/modules/adjudication/pages/site-overview/site-overview.component.ts:165]: Health Authority Site Overview ✔️ 
Health Authority Site Events ✔️ 
Health Authority Site Documents ✔️ ⚠️ I can upload the documents, but I can't see the documents anywhere except in the upload document page, is that expected behavior?
Health Authority Site Notes ✔️ 
Health Authority Site Actions ✔️ ⚠️ I can see the actions, however I cannot approve the site. I am not sure if it is because we don't have the field to assign Site ID such as the one we have for community sites where we have to provide a Site ID before we can approve it. If make the site editable and login as the authorized user and edit the site ID, then going back as an admin I can approve the site.
Site Overview ✔️ 
Site Actions ✔️

Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/pages/site-overview/site-overview.component.ts:165]: Health Authority Site Documents ✔️ ⚠️ I can upload the documents, but I can't see the documents anywhere except in the upload document page, is that expected behavior?

Yes

Actions: I have moved the checks so only community sites approvals are blocked missing a Pec. There was no mention of being able to assign pec/siteid for admin for ha sites so that might be flushed out later.

**Review by AMiKaH — APPROVED** [review]

File: prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs:271]: do we still need this comment or the one a couple lines up?

File: prime-dotnet-webapi/Controllers/SitesController.cs
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SitesController.cs:708]: probably should be `EditSite`

File: prime-dotnet-webapi/Controllers/SitesController.cs
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SitesController.cs:877]: So no emails are send when a HA site is approved?

File: prime-dotnet-webapi/Controllers/SitesController.cs
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SitesController.cs:862]: This method was updated to handle `CommunitySite` vs `HealthAuthoritySite` but here it is only being called for `CommunnitySite`s

File: prime-dotnet-webapi/Controllers/SitesController.cs
Comment (hogmann) [reply prime-dotnet-webapi/Controllers/SitesController.cs:877]: There is a ticket that has some email work in it, that still needs PO involvement

File: prime-dotnet-webapi/Controllers/SitesController.cs
Comment (hogmann) [reply prime-dotnet-webapi/Controllers/SitesController.cs:862]: I can revert the work to this call so it's just still a community site.

File: prime-dotnet-webapi/Services/SiteService.cs
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/SiteService.cs:50]: This needs `.DecompileAsync`, `site.Status` is a computed property

File: prime-dotnet-webapi/Services/SiteService.cs
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/SiteService.cs:43]: `Site.Status` is actually _current status_, can we make this name reflect that?

**Review by Paahn — APPROVED** [review]
functionality, code and tests LGTM

File: prime-angular-frontend/src/app/shared/components/health-auth-site-overview-container/health-auth-site-overview-container.component.html
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/health-auth-site-overview-container/health-auth-site-overview-container.component.html:6]: Accessing the `careType` in the care type overview needs a null check `?.careType`.  If you create a Health Authority Site, but only save the first page and exit then in admin visit the only partially filled out site (only vendor) then you get the infinite dialog loop.

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.html
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.html:24]: Noticed that if I remove notice it performs the action properly, but the red notification icon doesn't get removed until you refresh the page.

Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.html:24]: looks like i missed watching the reload event!

**Review by mtpultz — APPROVED** [review]

**Review by james-hollinger — APPROVED** [review]
LGTM

**Pull request merged**
